### PR TITLE
Updated release notes for 1.20

### PR DESCRIPTION
**Pull request by XAMPPRocky**
[Rendered](https://github.com/Aaronepower/rust/blob/master/RELEASES.md)

### CODE REVIEW

**Comment by wesleywiser** [comment]
The `compile_error!()` macro was also stabilized in `1.20` see #40872

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,140 @@
...
+- [Changed the convention around which file is the crate root.][cargo/4259]
+- [The `include`/`exclude` property in `Cargo.toml` now accepts gitignore paths
+  instead of glob patterns][cargo/4270]. Glob patterns are now deprecated.
+
+Compatibility Notes
+-------------------
+
+- [Functions with `'static` in their return types will now not be as usable as
+  if they were using lifetime parameters instead.][42417]
+
```
Comment (kennytm) [inline RELEASES.md]: #42431 should be a compatibility note. Previously NaN returns false for both `is_sign_positive` and `is_sign_negative`, and now they take the sign of NaN into account (following IEEE-754 definition). Code relying on the old behavior will be broken.

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,140 @@
...
+- [Impl `fmt::{Display, Debug}` for `Ref`, `RefMut`, `MutexGuard`,
+  `RwLockReadGuard`, `RwLockWriteGuard`][42822]
+- [Impl `Clone` for `DefaultHasher`.][42799]
+- [Impl `Sync` for `SyncSender`.][42397]
+- [Impl `FromStr` for `char`][42271]
+- [Fixed how `{f32, f64}::{is_sign_negative, is_sign_positive}` handles
+  NaN.][42431]
+- [allow messages in the `unimplemented!()` macro.][42155]
+  ie. `unimplemented!("Waiting for 1.21 to be stable")`
+- [support pub(restricted) in thread_local! (round 2)][43185]
```
Comment (kennytm) [inline RELEASES.md]: No need to say "round 2" :)

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,140 @@
...
+- [Upgrade to Unicode 10.0.0][42999]
+- [Reimplemented `{f32, f64}::{min, max}` in Rust instead of using CMath.][42430]
+- [Skip the main thread's manual stack guard on Linux][43072]
+- [Iterator::nth for `ops::{Range, RangeFrom}` is now done in O(1) time][43077]
+- [`#cfg[align(N)]` attribute max number is now 2^31 - 1.][43097] This was
```
Comment (kennytm) [inline RELEASES.md]: `#[repr(align(N))]`, not "cfg".

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,140 @@
...
+
+Language
+--------
+- [Associated constants in traits is now stabilised.][42809]
+- [A lot of macro bugs are now fixed.][42913]
+
+Compiler
+--------
+
+- [Struct fields are now properly to the expected field type.][42807]
```
Comment (KalitaAlexey) [inline RELEASES.md]: I believe it should be "Struct fields are now properly coerced..."

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,187 @@
...
+- [`str::get`]
+- [`str::into_boxed_bytes`]
+
+
+Cargo
+-----
+- [Cargo API token location moved from `~/.cargo/config` to
+  `~/cargo/credentials`.][cargo/3978]
+- [Cargo will now build multiple `main.rs` binaries in different
+  directories.][cargo/4214] ie. Having `src/server/main.rs` and
```
Comment (alexcrichton) [inline RELEASES.md]: Oh this is actually a little different than mentioned here. The change here was to infer `src/bin/$binary_name/main.rs` automatically as producign a `$binary_name` binary. Previously we'd just recognie `src/bin/*.rs`, but now we're also recognizing `src/bin/*/main.rs`

**Comment by alexcrichton** [comment]
Looking great tome, thanks so much again for tackling this @Aaronepower!

I don't mind r+'ing soon in the release cycle this time, that way we can edit it in-tree via PRs!

**Comment by Aaronepower** [comment]
@alexcrichton I've updated the wording of the Cargo change. Let me know if it's contains any more inaccuracies!

**Comment by alexcrichton** [comment]
@bors: r+

Looks great! Thanks again @Aaronepower!

**Pull request merged**